server/mailermost/mailermost.go

```diff
@@ -75,20 +74,25 @@ func (r *replyToBatchError) Error() string {
 	return r.Message
 }
 
-func (p *Poller) checkMailbox() {
+func (p *Poller) checkMailbox() error {
 	c, err := client.DialTLS(p.server, nil)
 	if err != nil {
-		p.api.LogError(fmt.Sprintf("failure connecting to IMAP server: %s", err.Error()))
+		return errors.Wrap(err, "failure connecting to IMAP server")
 	}
 
-	if err := c.Login(p.email, p.password); err != nil {
-		p.api.LogError(fmt.Sprintf("failure loging into email for user %q: %s", p.email, err.Error()))
+	if err = c.Login(p.email, p.password); err != nil {
+		return errors.Wrapf(err, "failure loging into email for user %q", p.email)
```

Comment:
I did refactor the code a bit to make it more readable and align the style with other repos